src/JsonValidator.php

```diff
@@ -35,6 +35,33 @@ class JsonValidator
     private $metaSchema;
 
     /**
+     * Callable creating new Validator instances.
+     *
+     * @var callable
+     */
+    private $validatorFactory;
+
+    /**
+     * JsonValidator constructor.
+     *
+     * @param callable|null $validatorFactory Factory creating JsonSchema\Validator instances.
```

Comment:
Why do you need support for callables in here instead of directly injecting the Validator?